internal/config/image/builder.go

```diff
@@ -24,9 +24,12 @@ import (
 )
 
 type builder struct {
+	spec           *specs.Spec
```

Comment:
I don't think that we should include a reference to the spec here -- or if we do we should remove the `env` and `mounts` below since they contain the same information -- or if they don't it is not clear which should take precedence.